providers/gitlab.go

```diff
@@ -206,15 +228,67 @@ func (p *GitLabProvider) EnrichSessionState(ctx context.Context, s *sessions.Ses
 		return fmt.Errorf("user email is not verified")
 	}
 
-	// Check group membership
-	// TODO (@NickMeves) - Refactor to Authorize
-	err = p.verifyGroupMembership(userInfo)
-	if err != nil {
-		return fmt.Errorf("group membership check failed: %v", err)
-	}
+	p.addGroupMembershipToState(ctx, s)
+
+	p.addProjectMembershipToState(ctx, s)
 
 	s.User = userInfo.Username
 	s.Email = userInfo.Email
 
 	return nil
+
+}
+
+// addProjectMembership adds projects into session.Groups
+func (p *GitLabProvider) addGroupMembershipToState(ctx context.Context, s *sessions.SessionState) {
+	if len(p.Groups) == 0 {
```

Comment:
Is this needed? If empty won't the for loop on the next line just do nothing anyway?

Reply:
No, a rest from the old, fixed : https://github.com/oauth2-proxy/oauth2-proxy/pull/630/commits/c2208668dc8cff916a5a1c811b99391612ac88b9